hazelcast/src/main/java/com/hazelcast/nio/Bits.java

```diff
@@ -337,4 +348,34 @@ public static long combineToLong(int x, int y) {
     public static int extractInt(long value, boolean lowerBits) {
         return (int) ((lowerBits) ? value : (value >> 32));
     }
+
+    /**
+     * Fast method of finding the next power of 2 greater than or equal to the supplied value.
+     * <p/>
+     * If the value is &lt;= 0 then 1 will be returned.
+     * <p/>
+     * This method is not suitable for {@link Integer#MIN_VALUE} or numbers greater than 2^30.
+     *
+     * @param value from which to search for next power of 2
+     * @return The next power of 2 or the value itself if it is a power of 2
+     */
+    public static int findNextPositivePowerOfTwo(final int value) {
```

Comment:
There two versions of `nextPowerOfTwo()` one is int-based, other is long. `public static long nextPowerOfTwo(long value)` and `public static int nextPowerOfTwo(int value)`. I don't say this one is better, I just say we should have only a single version of next-power-of-two.

Also what do you expect `nextPowerOfTwo(1L << 63)` to return? A positive value?  `1L << 63` itself is a negative value.